---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "image/exampple-",
  out.width = "100%"
)
```

# PureMeta <img src='image/PureMeta_logo.jpg' align="right" height="139" /></a>


<!-- badges: start -->
<!-- badges: end -->

## Overview

The goal of **PureMeta** is to provide functionality for extracting tumor cells-based gene expression profiles from bulk RNA-seq or microarray data, and generating tumor metabolic phenotype (i.e. downregulated, neutral, upregulated) for seven metabolic pathways.

## Installation

You can install the development version of PureMeta from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("WangKang-Leo/PureMeta")
```

## Example

This is a basic example which shows you how to solve a common problem:

```{r example}
library(PureMeta)
## basic example code
```

What is special about using `README.Rmd` instead of just `README.md`? You can include R chunks like so:

```{r cars}
summary(cars)
```

You'll still need to render `README.Rmd` regularly, to keep `README.md` up-to-date. `devtools::build_readme()` is handy for this. You could also use GitHub Actions to re-render `README.Rmd` every time you push. An example workflow can be found here: <https://github.com/r-lib/actions/tree/v1/examples>.

You can also embed plots, for example:

```{r pressure, echo = FALSE}
plot(pressure)
```

In that case, don't forget to commit and push the resulting figure files, so they display on GitHub and CRAN.

## Citation

If PureMeta software is used in your publication, please cite the following paper(s): 

* Longitudinal molecular profiling elucidates immunometabolism dynamics in breast cancer. [Kang Wang et al., Nature Communications 2023]() . (application of PureMeta to the longitudinal breast cancer cohort).

